backend/de.metas.business/src/main/java/de/metas/costing/CostAmount.java

```diff
@@ -178,18 +187,41 @@ else if (percent.isOneHundred())
 
 	public CostAmount add(@NonNull final CostAmount amtToAdd)
 	{
-		assertCurrencyMatching(amtToAdd);
+		final CostAmount toAddInSameCurrency;
+		if (!currencyId.equals(amtToAdd.currencyId))
+		{
+			toAddInSameCurrency = convertToSameCurrency(amtToAdd);
+		}
+		else
+		{
+			toAddInSameCurrency = amtToAdd;
+		}
+		assertCurrencyMatching(toAddInSameCurrency);
 
-		if (amtToAdd.isZero())
+		if (toAddInSameCurrency.isZero())
 		{
 			return this;
 		}
 		if (isZero())
 		{
-			return amtToAdd;
+			return toAddInSameCurrency;
 		}
 
-		return new CostAmount(value.add(amtToAdd.value), currencyId);
+		return new CostAmount(value.add(toAddInSameCurrency.value), currencyId);
+	}
+
+	@NonNull
+	private CostAmount convertToSameCurrency(final @NonNull CostAmount amtToAdd)
+	{
+		final CostAmount toAddInSameCurrency;
+		final CurrencyConversionContext currencyConversionContext = currencyBL.createCurrencyConversionContext(
+				SystemTime.asLocalDate(),
+				ConversionTypeMethod.Spot,
+				Env.getClientId(),
+				Env.getOrgId());
```

Comment:
please consult with teo on this. my 5cents:
- the method should be taken from the masterdata (i think teo, cosmin or florina shold know where it's set)
- clientId and orgId shoul be taken noch from the CTX but from the actual document or costing-record that is updated

to achieve this, a good way might be to introduce a class like `AddCostAmountRequest` where all these infos are contained